### ISSUE TITLE
concourse worker making excessive writes to disk

### ISSUE THREAD

**Issue by natalieparellano** [concourse/concourse#2906]
## Bug Report

We have a Linux worker running in vsphere that seems to be making excessive writes to disk - on average of 50,000 Kbps. Redeploying the worker fixed the issue for perhaps 24 hours (bringing the disk iops down to ~7,000 Kbps) before it spiked up again.

## Steps to Reproduce

We are using the following opsfiles from concourse-bosh-deployment:

cluster/external-worker.yml
cluster/operations/worker-ephemeral-disk.yml
cluster/operations/windows-worker.yml
cluster/operations/windows-worker-ephemeral-disk.yml

At some point, we removed our Windows worker to see if that was making a difference, but the high iops persist.

When we ssh onto the worker, using `iotop` we see that the offending process appears to be using the `loop0` device:

`# iotop -o -d 5 -a`
![worker-iops](https://user-images.githubusercontent.com/6894483/49404156-a38ef480-f71c-11e8-8dfa-8d1618c85e5d.png)

Also relevant, `loop0` is being used by baggageclaim:

```
$ mount | grep loop0
/dev/loop0 on /var/vcap/data/baggageclaim/volumes type btrfs (rw,relatime,space_cache,subvolid=5,subvol=/)
```

## Expected Results

50,000 Kbps seems very high. We were expecting lower iops activity. This worker is only running a `cf push` every 5 minutes and a `curl` every 1 minute. It's hard to imagine what all it could be writing to disk!

## Version Info

* Concourse version: 4.2.1
* Deployment type (BOSH/Docker/binary): bosh
* Infrastructure/IaaS: vsphere
* Browser (if applicable):
* Did this used to work? ~~We don't know :(~~ Actually this **did** used to work - the worker was up for 30+ days before we started seeing the issue (judging from performance charts in vCenter). However when we did a fresh redeploy, the issue reoccurred within ~24 hours.

**Comment by natalieparellano** [comment]
Some more things we learned:

* We see a bunch of log lines with `baggageclaim.repository.destroy-volume.failed-to-destroy` and a bunch of dead containers in `/var/vcap/data/baggageclaim/volumes/dead` - maybe this is related to our issue? We unfortunately don't have the logs for the time period where we also know exactly when the spike in disk writes started. 
* We don't see excessive log lines printed in `baggageclaim.stdout.log` even when the spike in disk writes is occurring (the time period covered by each log roughly matches what we see when there is no spike)

Can you suggest other data we should be collecting, or other things to look for? We are starting to impact other teams in the data center with our excessive disk usage so we would like to get to the bottom of this.

**Comment by samgurtman** [comment]
We are having somewhat similar issues on 4.2.2. I'm not sure if it's the same but we also get various failed-to-destroy and other volume management errors. This causes the worker to become stalled.

**Comment by taylorsilva** [comment]
# Issue Bankruptcy

This issue is being closed as part of 2025's **Issue Bankruptcy** declaration. [See here for details](https://blog.concourse-ci.org/posts/2025-01-06-concourse-2024-review/)

If you think this issue still has merit then feel free to open a new issue and reference this one.
